WordPress/Classes/ViewRelated/Views/WPRichTextGallery.swift

```diff
@@ -0,0 +1,335 @@
+import UIKit
+import WordPressShared
+
+class WPRichTextGallery: UIView, WPRichTextMediaAttachment {
+
+    /// MARK - Constants
+    ///
+
+    fileprivate let galleryCellIdentifier = "ReaderGalleryCell"
+    fileprivate let contentUrlKey = "contentUrl"
+    fileprivate let linkUrlKey = "linkUrl"
+    fileprivate let viewHeightKey = "viewHeight"
+
+    fileprivate var viewHeight: CGFloat
+
+    var isPrivate = false
+    var contentURL: URL?
+    var linkURL: URL?
+
+    /// Default cell size, will be recalculated as page is laid out
+    ///
+    var cellWidth: CGFloat = 1.0
+    let cellPadding: CGFloat = 8.0
+
+    /// Used to load images for attachments.
+    ///
+    lazy var imageSource: WPTableImageSource? = {
+        let source = WPTableImageSource(maxSize: self.maxDisplaySize)
+        source?.delegate = self
+        source?.forceLargerSizeWhenFetching = false
+        source?.photonQuality = 65
+        return source
+    }()
+
+    fileprivate var collectionView: UICollectionView
+    fileprivate var footerStack: UIStackView
+
+    var captionLabel: UILabel?
+    var pageLabel: UILabel?
+
+    /// Used to keep references to image attachments.
+    ///
+    fileprivate var mediaArray = [GalleryMedia]()
+
+    /// The maximum size for images.
+    ///
+    lazy var maxDisplaySize: CGSize = {
+        let bounds = UIScreen.main.bounds
+        let side = max(bounds.size.width, bounds.size.height)
+        return CGSize(width: side, height: side)
+    }()
+
+    let layout: UICollectionViewFlowLayout = {
+        let layout = UICollectionViewFlowLayout()
+        layout.scrollDirection = .horizontal
+        layout.minimumInteritemSpacing = 8.0
```

Comment:
We could use a constant for these values – or should this be using `cellPadding`?